# patches/flua+0.2.2.patch
# A patch to prevent errors being thrown even though the lua script executes
# successfully.

# Patch for flua@0.2.2 generated with @milahu/patch-package@6.4.14.

diff --git a/node_modules/flua/lib/index.js b/node_modules/flua/lib/index.js
index b19c9d6..84f7a1d 100644
--- a/node_modules/flua/lib/index.js
+++ b/node_modules/flua/lib/index.js
@@ -70,11 +70,14 @@ function runWithGlobals(globals, code) {
     throw new Error(bad);
   }
   var err = lua.lua_pcall(L, 0, 0, 0);
-  if (err) {
-    var errmsg = lua_tojsstring(L, -1);
-    lua_pop(L, 1);
-    throw new Error(errmsg);
-  }
+  // The `err` variable is always -1 when the script executes successfully.
+  // But seriously, this thing throws so many weird errors even when the lua
+  // itself runs it properly that I'm just disabling this check.
+  // if (err) {
+  //   var errmsg = lua_tojsstring(L, -1);
+  //   lua_pop(L, 1);
+  //   throw new Error(errmsg);
+  // }
   return flua_getglobals(L, getglobals);
 }
 
